qiskit-superstaq/qiskit_superstaq/superstaq_provider.py

```diff
@@ -128,11 +128,12 @@ def backends(self) -> List[qss.SuperstaqBackend]:
         Returns:
             A list of Superstaq backends.
         """
-        targets = self._client.get_targets()["superstaq_targets"]
-        backends = []
-        for target in targets["compile-and-run"]:
-            backends.append(self.get_backend(target))
-        return backends
+        submit_targets = self._client.get_targets()
+        superstaq_backends = []
+        for backend in submit_targets:
+            if backend.supports_submit:
+                superstaq_backends.append(self.get_backend(backend.target))
```

Comment:
predates this pr, but now that we support `compile()` and simulation for any target we shouldn't be restricting this to those that support submission
```suggestion
        targets = self._client.get_targets()
        superstaq_backends = []
        for backend in targets:
            superstaq_backends.append(self.get_backend(backend.target))
```